.pre-commit-config.yaml

```diff
@@ -15,3 +15,7 @@ repos:
   hooks:
   - id: isort
     name: isort (python)
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.4.9
+  hooks:
+    - id: ruff # run the linter
```

Comment:
imo we can just drop isort and black entirely in favour of ruff check + format - would you mind adding that to this PR?

Reply:
Also very small nit but if we drop isort and black we should update the badges at the top of the README :)

Reply:
god point, I'll see to it